gradle.properties

```diff
@@ -1,2 +1,3 @@
 org.gradle.parallel=true
 archunit.version=0.23.0-SNAPSHOT
+org.gradle.jvmargs=--add-exports jdk.compiler/com.sun.tools.javac.code=ALL-UNNAMED --add-exports jdk.compiler/com.sun.tools.javac.file=ALL-UNNAMED --add-exports jdk.compiler/com.sun.tools.javac.parser=ALL-UNNAMED --add-exports jdk.compiler/com.sun.tools.javac.tree=ALL-UNNAMED --add-exports jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED --add-exports jdk.compiler/com.sun.tools.javac.api=ALL-UNNAMED
```

Comment:
Why do we need this only now, but not before? :thinking: Is it related to JDK 8 or JDK 17 or upgrading Gradle? In any case an explanation in the respective commit message would be cool :slightly_smiling_face:

Reply:
These were successively encountered when running on JDK 17; they are needed since JDK 17 further restricts access to internals. For instance Spotless fails with a message such as:
```
Caused by: java.lang.IllegalAccessError: class com.google.googlejavaformat.java.RemoveUnusedImports (in unnamed module @0x3e573cb9) cannot access class com.sun.tools.javac.util.Context (in module jdk.compiler) because module jdk.compiler does not export com.sun.tools.javac.util to unnamed module @0x3e573cb9
        at com.google.googlejavaformat.java.RemoveUnusedImports.removeUnusedImports(RemoveUnusedImports.java:188)
        ... 139 more
```

Reply:
Ah, okay :+1: Then probably would also be good to add that explanation to the commit message of the JDK upgrade commit :slightly_smiling_face:

Reply:
Pushed separately with comment.